client/src/pages/SinglePost.jsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useEffect } from "react";
+import { useParams, Link, useNavigate } from "react-router-dom";
+
+const SinglePost = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+
+  const upload = async () => {
+    try {
+      const formData = new FormData();
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const [singlePost, setSinglePost] = useState();
+  const getSinglePost = async () => {
+    try {
+      const response = await fetch(`http://localhost:8080/api/v1/posts/${id}`);
+
+      if (response.ok) {
+        const {
+          data: { post },
+        } = await response.json();
+        setSinglePost(post[0]);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const deletePost = async () => {
+    try {
+      const response = await fetch(`http://localhost:8080/api/v1/posts/${id}`, {
+        method: "DELETE",
+        headers: {
+          "Content-type": "application/JSON",
+        },
+      });
+
+      if (response.ok) {
+        const data = await response.json();
+        navigate("/");
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  console.log(singlePost);
+  useEffect(() => {
+    getSinglePost();
+  }, []);
+
+  return (
+    singlePost && (
+      <div className="singlePost">
+        <div className="singlePostInfo">
+          <div className="singleImgContainer">
+            <img
+              className="singleImg"
+              src={`../../public/uploads/${singlePost.img}`}
+            ></img>
+          </div>
+          <div className="btnContainer">
+            <Link className="editBtn" to="/write" state={singlePost}>
```

Comment:
This feels a little weird to me. I think I'd make this component handle the editing itself, rather than routing to another location.

Reply:
I would like to talk about this one further!